theories/datatypes/List.ec

```diff
@@ -1818,6 +1819,137 @@ lemma mapiP x0 (f : int -> 'a -> 'b) (s : 'a list) y :
     exists n, (0 <= n && n < size s) /\ y = f n (nth x0 s n).
 proof. exact: mapi_recP. qed.
 
+(* -------------------------------------------------------------------- *)
+(*                          Element Replacement                         *)
+(* -------------------------------------------------------------------- *)
+op put (s : 'a list) (n : int) (x : 'a) =
+  if (0 <= n < size s)
+  then take n s ++ (x :: drop (n + 1) s)
+  else s.
+
+lemma put_empty (n : int) (x : 'a) :
+  put [] n x = [].
+proof. by smt(). qed.
+
+lemma size_put (s : 'a list) (n : int) (x : 'a) :
+  size (put s n x) = size s.
+proof.
+rewrite /put; case (0 <= n < size s) => [[? lts_n] | //].
+by rewrite size_cat size_take // lts_n /= size_drop /#.
+qed.
+
+lemma put_in (s : 'a list) (n : int) (x : 'a) :
+  0 <= n < size s => put s n x = take n s ++ (x :: drop (n + 1) s).
+proof. by rewrite /put => ->. qed. 
+
+lemma put_out (s : 'a list) (n : int) (x : 'a) :
+  !(0 <= n < size s) => put s n x = s.
+proof. by rewrite /put => ->. qed. 
+
+lemma put_neg (s : 'a list) (n : int) (x : 'a) :
+  n < 0 => put s n x = s.
+proof. by move => ?; rewrite put_out /#. qed.
+
+lemma put_oversize (s : 'a list) (n : int) (x : 'a) :
+  size s <= n => put s n x = s.
+proof. by move => ?; rewrite put_out /#. qed.
```

Comment:
I think `put_out` is enough, in particular since one can do `rewrite put_out 1:/#`